src/types/common/token.ts

```diff
@@ -1,8 +1,59 @@
+export const TokenTypes = {
+	OAUTH_ACCESS: 1,
+	OAUTH_REFRESH: 2,
+	NEX: 3,
+	SERVICE: 4,
+	PASSWORD_RESET: 5
+} as const;
+export type TokenType = keyof typeof TokenTypes;
+
+export function getTokenTypeFromValue(type: number): TokenType | undefined {
+	const keys = Object.keys(TokenTypes) as TokenType[];
+	return keys.find((key) => TokenTypes[key] === type);
+}
+
+export const SystemTypes = {
+	'WIIU': 1,
+	'3DS': 2,
+	'API': 3
+} as const;
+export type SystemType = keyof typeof SystemTypes;
+
+export function getSystemTypeFromValue(type: number): SystemType | undefined {
+	const keys = Object.keys(SystemTypes) as SystemType[];
+	return keys.find((key) => SystemTypes[key] === type);
+}
```

Comment:
The `types` folder is really only supposed to be for type defs, I'm not sure we should be putting helper functions in here?

Reply:
It's something I do to keep them local to the things they are relevant to - since they aren't generic helper methods